src/Coordination/SnapshotableHashTable.h

```diff
@@ -288,6 +363,7 @@ class SnapshotableHashTable
 
     void clear()
     {
+        clearOutdatedNodes();
```

Comment:
this function seems to clear all the map. Why clear outdated nodes first?

Reply:
because the key is shared so we don't want to deallocate the same key twice.
We first clear all duplicates created for snapshot with `clearOutdatedNodes`  and then all the nodes.